libgobuster/statehelpers.go

```diff
@@ -170,6 +171,22 @@ func ValidateDirModeState(
 		}
 	}
 
+	// Read the user agents file
+	if errorList.ErrorOrNil() == nil && s.RandomUserAgent {
+		b, err := ioutil.ReadFile(s.UserAgentsFile)
+		if err != nil {
+			errorList = multierror.Append(errorList, fmt.Errorf("[!] Error when reading the User-Agents file"))
```

Comment:
We're masking the cause of the error here by not including it. Could just add the error directly to the errorList, though then you can't add your additional context. At the very least I would include the error message with something like:

```
fmt.Errorf("[!] Error when reading the User-Agents file: %s", err)
```

Even that isn't super ideal. Could potentially use `%+v` to get more context in it? In my own code, I normally would wrap it with [another library](https://github.com/pkg/errors):

```
errors.Wrap(err, "[!] Error when reading the User-Agents file")
```

That would be my best case scenario to do things 'right', but would require adding in another dependency (which personally I don't see as a huge issue).

Reply:
I like the `%+v`:

```./gobuster -u https://google.com/ -w <(echo "images") -random-agent

Gobuster v1.4.1              OJ Reeves (@TheColonial)
=====================================================
1 error occurred:

* [!] Error when reading the User-Agents file: open user-agents.txt: no such file or directory```